### PR TITLE
feat(Selectdropdown):  Add subtitle, rightlabel, disabled, and divider options

### PR DESCRIPTION
**Pull request by CharlieBliss**
### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: [ABC-123]
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change
- [ ] This PR includes testing instructions tests for the code change
- [ ] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### Testing Instructions

- View the SelectDropdown Option Variants section
- The first dropdown should have disabled options
- They should be focusable and should be read as disabled by screenreaders, but should not be selectable or hoverable

- The second dropdown should have subtitles, right-aligned labels, a divider, and have some disabled options
- The disabled options should have all text colored with text-disabled

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](http://www.google.fr/ 'Named link title')  | [Monolith Env](http://www.google.fr/ 'Named link title') |
| Portal       | [Portal Link](http://www.google.fr/ 'Named link title')  | [Portal Env](http://www.google.fr/ 'Named link title')   |
| Another Repo | [Another Link](http://www.google.fr/ 'Named link title') | [Another Env](http://www.google.fr/ 'Named link title')  |

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.205-alpha.33659b.0<br/>@codecademy/gamut-labs@33.5.1-alpha.33659b.0<br/>@codecademy/gamut@50.6.6-alpha.33659b.0<br/>@codecademy/monaco@0.6.88-alpha.33659b.0<br/>@codecademy/styleguide@61.5.12-alpha.33659b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://632b414a6a78db6ffb7091ba--gamut-preview.netlify.app>https://632b414a6a78db6ffb7091ba--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/632b414a6a78db6ffb7091ba>Deploy Logs</a></p>

**Review by dreamwasp — COMMENTED** [review]
did some really quick testing on this + noticed it doesn't appear that you can select the options on the rightLabel/extended options example:

https://user-images.githubusercontent.com/58054751/191800914-088aa651-8a87-4694-85b2-79081b587d49.mov

**Comment by CharlieBliss** [comment]
> did some really quick testing on this + noticed it doesn't appear that you can select the options on the rightLabel/extended options example:
> 
>  Screen.Recording.2022-09-22.at.9.21.25.AM.mov

Yeah this is because of some custom logic happening around option parsing -- updating to support grouped options

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.205-alpha.5c19e5.0<br/>@codecademy/gamut-labs@33.5.1-alpha.5c19e5.0<br/>@codecademy/gamut@50.6.6-alpha.5c19e5.0<br/>@codecademy/monaco@0.6.88-alpha.5c19e5.0<br/>@codecademy/styleguide@61.5.12-alpha.5c19e5.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.205-alpha.b090b8.0<br/>@codecademy/gamut-labs@33.5.1-alpha.b090b8.0<br/>@codecademy/gamut@50.6.6-alpha.b090b8.0<br/>@codecademy/monaco@0.6.88-alpha.b090b8.0<br/>@codecademy/styleguide@61.5.12-alpha.b090b8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6331d7fae30be70c843cc7a4--gamut-preview.netlify.app>https://6331d7fae30be70c843cc7a4--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6331d7fae30be70c843cc7a4>Deploy Logs</a></p>

**Review by dreamwasp — COMMENTED** [review]
could you add the alpha pkg to the monolith + link it here? just wanna double-check the existing usages

**Comment by CharlieBliss** [comment]
@dreamwasp 
https://bantam.codecademy.com/ has the alpha package

**Comment by dreamwasp** [comment]
> @dreamwasp https://bantam.codecademy.com/ has the alpha package

cool - looks good there! i also just wanna double check all the e2e tests pass

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.205-alpha.eefb95.0<br/>@codecademy/gamut-labs@33.5.1-alpha.eefb95.0<br/>@codecademy/gamut@50.6.6-alpha.eefb95.0<br/>@codecademy/monaco@0.6.88-alpha.eefb95.0<br/>@codecademy/styleguide@61.5.12-alpha.eefb95.0<br/></pre>

**Comment by CharlieBliss** [comment]
> > @dreamwasp https://bantam.codecademy.com/ has the alpha package
> 
> cool - looks good there! i also just wanna double check all the e2e tests on the monolith + the monorepo still pass

Seems chill 

https://github.com/codecademy-engineering/Codecademy/pull/32794

File: packages/styleguide/stories/Atoms/FormInputs/SelectDropdown.stories.mdx [hunk 1]
```diff
@@ -369,6 +373,127 @@ You can chose how options will be shown in the dropdown. 6 is the default limit
...
+              ],
+            },
+          ]}
+      />
+    </FormGroup>
+  </Box>
+);
+```
+
+### SelectDropdown Multiple Select
```
Comment (dreamwasp) [inline packages/styleguide/stories/Atoms/FormInputs/SelectDropdown.stories.mdx:461]: there's small visual bug here - 
image

File: packages/styleguide/stories/Atoms/FormInputs/SelectDropdown.stories.mdx [hunk 2]
```diff
@@ -369,6 +373,127 @@ You can chose how options will be shown in the dropdown. 6 is the default limit
   </Story>
 </Canvas>
 
+### SelectDropdown Option Variants
+
+You may need to add additional info to your dropdowns, like disabled options, option subtitles, or group dividers
```
Comment (dreamwasp) [inline packages/styleguide/stories/Atoms/FormInputs/SelectDropdown.stories.mdx:378]: disabled options currently don't have a disabled cursor on hover - was this a deliberate choice?

**Review by dreamwasp — APPROVED** [review]
couple of small comments but otherwise these changes look good to me!

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://633aeba6b0b2e419542f8b09--gamut-preview.netlify.app>https://633aeba6b0b2e419542f8b09--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/633aeba6b0b2e419542f8b09>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.208-alpha.c52dcb.0<br/>@codecademy/gamut-labs@33.5.4-alpha.c52dcb.0<br/>@codecademy/gamut@50.6.7-alpha.c52dcb.0<br/>@codecademy/monaco@0.6.89-alpha.c52dcb.0<br/>@codecademy/styleguide@61.5.15-alpha.c52dcb.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://633c6f02f6573358eaa246d3--gamut-preview.netlify.app>https://633c6f02f6573358eaa246d3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/633c6f02f6573358eaa246d3>Deploy Logs</a></p>

File: packages/styleguide/stories/Atoms/FormInputs/SelectDropdown.stories.mdx (hunk 2 above)
Comment (CharlieBliss) [reply packages/styleguide/stories/Atoms/FormInputs/SelectDropdown.stories.mdx:378]: no -added

File: packages/styleguide/stories/Atoms/FormInputs/SelectDropdown.stories.mdx (hunk 1 above)
Comment (CharlieBliss) [reply packages/styleguide/stories/Atoms/FormInputs/SelectDropdown.stories.mdx:461]: seems like this is an issue with the storybook container, I don't think it's really in the scope of this pr

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.208-alpha.34c415.0<br/>@codecademy/gamut-labs@33.5.4-alpha.34c415.0<br/>@codecademy/gamut@50.6.7-alpha.34c415.0<br/>@codecademy/monaco@0.6.89-alpha.34c415.0<br/>@codecademy/styleguide@61.5.15-alpha.34c415.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://633c739c0009c7004a1c7da0--gamut-preview.netlify.app>https://633c739c0009c7004a1c7da0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/633c739c0009c7004a1c7da0>Deploy Logs</a></p>

**Pull request merged**